### PR TITLE
Allow add-ons installation

### PR DESCRIPTION
**Pull request by tsl143**

### CODE REVIEW

File: mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm [hunk 1]
```diff
@@ -277,14 +277,24 @@ class Package {
         cert: null
       };
     }
+    /*
+      Ghostery - allow both Firefox and Cliqz certificate for installing addons
+    */
+    let root = Ci.nsIX509CertDB.AddonsPublicRoot;
```
Comment (luciancor) [inline mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm]: @tsl143 : we had a similar requirement on mobile, @moaz-cliqz implemented - https://github.com/ghostery/browser-android/pull/295/files

Not sure if required, but maybe worth using same implementation on both sides?

File: mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm [hunk 2]
```diff
@@ -1038,14 +1048,16 @@ function getSignedStatus(aRv, aCert, aAddonID) {
       if (expectedCommonName && expectedCommonName != aCert.commonName)
         return AddonManager.SIGNEDSTATE_BROKEN;
 
-      if (aCert.organizationalUnit == "Cliqz Frontend") {
+      if (aCert.organizationalUnit == "Cliqz Frontend" ||
```
Comment (luciancor) [inline mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm]: @tsl143 we need to block the instalation of the Mozilla signed addons which are also comming from Cliqz. EG: cliqz@cliqz.com, LastPass, 1Password and HttpsEverywhere

File: mozilla-release/toolkit/mozapps/extensions/content/extensions.js [hunk 3]
```diff
@@ -1644,6 +1648,12 @@ async function getAddonsAndInstalls(aType, aCallback) {
 
   let aAddonsList = await AddonManager.getAddonsByTypes(types);
   addons = aAddonsList.filter(a => !a.hidden);
+
+  if (Services.prefs.getPrefType("extensions.cliqz.listed")
+    && !Services.prefs.getBoolPref("extensions.cliqz.listed")) {
+      addons = addons.filter(isNotSystemExtension);
```
[the remark this answers is not in the document]
Comment (tsl143) [reply mozilla-release/toolkit/mozapps/extensions/content/extensions.js]: We are only hiding system addons, if user installs it manually he must see the addon in the listing.

File: mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm (hunk 1 above)
Comment (tsl143) [reply mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm]: yes, let's implement similar approach, I will update the PR

File: mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm (hunk 2 above)
Comment (tsl143) [reply mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm]: Okay, we need to put extra check for this.

**Comment by tsl143** [comment]
@luciancor @alver-cliqz , I have updated the PR, please review

File: mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm (hunk 1 above)
Comment (tsl143) [reply mozilla-release/toolkit/mozapps/extensions/internal/XPIInstall.jsm]: We need to do something different to consider the update system addon case, I have updated the PR with the same.

**Comment by luciancor** [comment]
LGTM!

**Pull request merged**